File: Heat/Controllers/CausalWarehousesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat
Imports Heat.Repositories

Namespace Controllers
    Public Class CausalWarehousesController
        Inherits System.Web.Mvc.Controller

        Private _db As IHeatDBContext

        Public Sub New(context As IHeatDBContext)
            _db = context
        End Sub

        ' GET: CausalWarehouses
        Function Index() As ActionResult
            Dim causalWarehouses = _db.CausalWarehouses.Include(Function(c) c.Type)
            Return View(causalWarehouses.ToList())
        End Function

        ' GET: CausalWarehouses/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim causalWarehouse As CausalWarehouse = _db.CausalWarehouses.Find(id)
            If IsNothing(causalWarehouse) Then
                Return HttpNotFound()
            End If
            Return View(causalWarehouse)
        End Function

        ' GET: CausalWarehouses/Create
        Function Create() As ActionResult
            ViewBag.TypeID = New SelectList(_db.CausalWarehouseGroups, "ID", "Code")
            Return View()
        End Function

        ' POST: CausalWarehouses/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,Code,Sign,TypeID,Transaction")> ByVal causalWarehouse As CausalWarehouse) As ActionResult
            If ModelState.IsValid Then
                _db.CausalWarehouses.Add(causalWarehouse)
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            ViewBag.TypeID = New SelectList(_db.CausalWarehouseGroups, "ID", "Code", causalWarehouse.TypeID)
            Return View(causalWarehouse)
        End Function

        ' GET: CausalWarehouses/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim causalWarehouse As CausalWarehouse = _db.CausalWarehouses.Find(id)
            If IsNothing(causalWarehouse) Then
                Return HttpNotFound()
            End If
            ViewBag.TypeID = New SelectList(_db.CausalWarehouseGroups, "ID", "Code", causalWarehouse.TypeID)
            Return View(causalWarehouse)
        End Function

        ' POST: CausalWarehouses/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,Code,Sign,TypeID,Transaction")> ByVal causalWarehouse As CausalWarehouse) As ActionResult
            If ModelState.IsValid Then
                _db.SetModified(causalWarehouse) '.State = EntityState.Modified
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            ViewBag.TypeID = New SelectList(_db.CausalWarehouseGroups, "ID", "Code", causalWarehouse.TypeID)
            Return View(causalWarehouse)
        End Function

        ' GET: CausalWarehouses/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim causalWarehouse As CausalWarehouse = _db.CausalWarehouses.Find(id)
            If IsNothing(causalWarehouse) Then
                Return HttpNotFound()
            End If
            Return View(causalWarehouse)
        End Function

        ' POST: CausalWarehouses/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim causalWarehouse As CausalWarehouse = _db.CausalWarehouses.Find(id)
            _db.CausalWarehouses.Remove(causalWarehouse)
            _db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                _db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/CausalWarehousesController.cs
using System.Linq;
using System.Web.Mvc;
using System.Data.Entity;
using System.Net;
using Heat.Models;

namespace Heat.Controllers
{
    public class CausalWarehousesController : System.Web.Mvc.Controller
	{


		private IHeatDBContext _db;
		public CausalWarehousesController(IHeatDBContext context)
		{
			_db = context;
		}

		// GET: CausalWarehouses
		public ActionResult Index()
		{
			var causalWarehouses = _db.CausalWarehouses;
			return View(causalWarehouses.ToList());
		}

		// GET: CausalWarehouses/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			CausalWarehouse causalWarehouse = _db.CausalWarehouses.Find(id);
			if ((causalWarehouse == null)) {
				return HttpNotFound();
			}
			return View(causalWarehouse);
		}

		// GET: CausalWarehouses/Create
		public ActionResult Create()
		{
			ViewBag.TypeID = new SelectList(_db.CausalWarehouseGroups, "ID", "Code");
			return View();
		}

		// POST: CausalWarehouses/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,Code,Sign,TypeID,Transaction")]
CausalWarehouse causalWarehouse)
		{
			if (ModelState.IsValid) {
				_db.CausalWarehouses.Add(causalWarehouse);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			ViewBag.TypeID = new SelectList(_db.CausalWarehouseGroups, "ID", "Code", causalWarehouse.TypeID);
			return View(causalWarehouse);
		}

		// GET: CausalWarehouses/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			CausalWarehouse causalWarehouse = _db.CausalWarehouses.Find(id);
			if ((causalWarehouse == null)) {
				return HttpNotFound();
			}
			ViewBag.TypeID = new SelectList(_db.CausalWarehouseGroups, "ID", "Code", causalWarehouse.TypeID);
			return View(causalWarehouse);
		}

		// POST: CausalWarehouses/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,Code,Sign,TypeID,Transaction")]
CausalWarehouse causalWarehouse)
		{
			if (ModelState.IsValid) {
				_db.SetModified(causalWarehouse);
				//.State = EntityState.Modified
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			ViewBag.TypeID = new SelectList(_db.CausalWarehouseGroups, "ID", "Code", causalWarehouse.TypeID);
			return View(causalWarehouse);
		}

		// GET: CausalWarehouses/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			CausalWarehouse causalWarehouse = _db.CausalWarehouses.Find(id);
			if ((causalWarehouse == null)) {
				return HttpNotFound();
			}
			return View(causalWarehouse);
		}

		// POST: CausalWarehouses/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			CausalWarehouse causalWarehouse = _db.CausalWarehouses.Find(id);
			_db.CausalWarehouses.Remove(causalWarehouse);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				_db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
